File: app/src/main/java/com/evans/technologies/conductor/ui/recover/recoverPass/correo.java
package com.evans.technologies.conductor.ui.recover.recoverPass;


import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ProgressBar;

import com.evans.technologies.conductor.R;
import com.evans.technologies.conductor.data.network.service.auth.ClienteRetrofit;
import com.evans.technologies.conductor.model.Driver;

import butterknife.BindView;
import butterknife.ButterKnife;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

import static com.evans.technologies.conductor.utils.UtilsKt.setIDrecuperar;
import static com.evans.technologies.conductor.utils.UtilsKt.setNavFragment;
import static org.jetbrains.anko.ToastsKt.toast;

/**
 * A simple {@link Fragment} subclass.
 */
public class correo extends Fragment {

    View view;
    SharedPreferences navFragment;
    @BindView(R.id.fc_btn_correo_ic)
    Button next;
    @BindView(R.id.c_edtxt_email_confirm)
    EditText email2;
    @BindView(R.id.c_edtxt_email)
    EditText email1;
    String e1,e2;
    @BindView(R.id.progressBar_correo)
    ProgressBar progressBar;
    public correo() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        navFragment= getContext().getSharedPreferences("navFragment", Context.MODE_PRIVATE);
        navFragment.edit().clear().apply();



        view= inflater.inflate(R.layout.fragment_correo, container, false);
        ButterKnife.bind(this,view);
        next.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                e1=email1.getText().toString().trim();
                e2=email2.getText().toString().trim();
                if (comprobarcampos()){
                    progressBar.setVisibility(View.GONE);
                    Call<Driver> sendCorreo= ClienteRetrofit.getInstance().getApi().sendCorreo_recuperar(e1);
                    sendCorreo.enqueue(new Callback<Driver>() {
                        @Override
                        public void onResponse(Call<Driver> call, Response<Driver> response) {
                            Log.e("correo_set",response.code()+"");
                            if (response.isSuccessful()){
                                if (response.body()!=null){
                                    progressBar.setVisibility(View.VISIBLE);
                                    Log.e("correo_set",response.body().getUser());
                                    setIDrecuperar(navFragment,response.body().getUser());
                                    setNavFragment(navFragment,new correo().toString());
                                    FragmentManager manager = getActivity().getSupportFragmentManager();
                                    manager.beginTransaction().replace(R.id.recuperar_frag,
                                            new set_codigo()
                                    ).commit();
                                }
                            }else{
                                progressBar.setVisibility(View.VISIBLE);
                                toast(getActivity(),"El correo no existe");
                            }
                        }

                        @Override
                        public void onFailure(Call<Driver> call, Throwable t) {
                            progressBar.setVisibility(View.VISIBLE);
                            Log.e("correo_set",t.getMessage()+"");
                        }
                    });
                }
            }


        });

        return view;
    }
    private boolean comprobarcampos() {
        if (e1.isEmpty()||e2.isEmpty()){
            email1.setError("Campos vacios");
            email2.setError("Campos vacios");
            return  false;
        }
        if (!(android.util.Patterns.EMAIL_ADDRESS.matcher(e1).matches())){
            email1.setError("No es un correo valido");
            return  false;
        }
        if (!(android.util.Patterns.EMAIL_ADDRESS.matcher(e2).matches())){
            email2.setError("No es un correo valido");
            return  false;
        }
        if (!e1.equals(e2)){
            email1.setError("Los correos no coinciden");
            email2.setError("Los correos no coinciden");
            return  false;
        }

        return true;
    }

}

File: app/src/main/java/com/evans/technologies/conductor/fragments/change_password/correo.kt
package com.evans.technologies.conductor.fragments.change_password

import android.content.Context
import android.content.SharedPreferences
import android.os.Bundle
import android.util.Log
import android.util.Patterns
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import com.evans.technologies.conductor.R
import com.evans.technologies.conductor.Retrofit.RetrofitClient
import com.evans.technologies.conductor.Utils.setCorreoNavFragment
import com.evans.technologies.conductor.Utils.setIDrecuperar
import com.evans.technologies.conductor.Utils.setNavFragment
import com.evans.technologies.conductor.model.Driver
import kotlinx.android.synthetic.main.fragment_correo.*
import org.jetbrains.anko.toast
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

/**
 * A simple [Fragment] subclass.
 */
class correo : Fragment() {
    lateinit var navFragment: SharedPreferences

    private var sendRecuperarContraseña=false
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        navFragment =
            context!!.getSharedPreferences("navFragment", Context.MODE_PRIVATE)
        navFragment.edit().clear().apply()
        val b= arguments
        if (b != null) {
            sendRecuperarContraseña = b.getBoolean("sendRecuperarContraseña")
        }

        fc_btn_correo_ic!!.setOnClickListener {
            val e1 = c_edtxt_email!!.text.toString().trim()
            val e2 = c_edtxt_email_confirm!!.text.toString().trim()
            if (comprobarcampos(e1,e2)) {
                Log.e("accountActivate",sendRecuperarContraseña.toString())
                if(sendRecuperarContraseña){
                    recuperarContraseña(e1)
                }else{
                    accountActivate(e1)

                }
            }
        }
    }
    fun recuperarContraseña(e1: String) {
        progressBar_correo!!.visibility = View.VISIBLE
        val sendCorreo =
            RetrofitClient.getInstance().api.sendCorreo_recuperar(e1!!)
        sendCorreo.enqueue(object : Callback<Driver?> {
            override fun onResponse(
                call: Call<Driver?>,
                response: Response<Driver?>
            ) {
                Log.e("correo_set", response.code().toString() + "")
                if (response.isSuccessful) {
                    progressBar_correo!!.visibility = View.GONE

                    Log.e("correo_set", response.body()!!.user)
                    setIDrecuperar(navFragment, response.body()!!.user)
                    setNavFragment(navFragment, correo().toString())
                    setCorreoNavFragment(navFragment, e1)
                    val bundle= Bundle()
                    bundle.putBoolean( "validar_account",false)
                    bundle.putBoolean("activity" ,false)

                    findNavController().navigate(R.id.action_correo_to_set_codigo,bundle)

                    //    findNavController().navigate(R.id.action_loginFragment_to_correo)

//                                val manager =
//                                    activity!!.supportFragmentManager
//                                manager.beginTransaction().replace(
//                                    R.id.recuperar_frag,
//                                    set_codigo(false, false)
//                                ).commit()

                } else {
                    progressBar_correo!!.visibility = View.GONE
                    activity!!.toast("El correo no existe")
                }
            }

            override fun onFailure(
                call: Call<Driver?>,
                t: Throwable
            ) {
                progressBar_correo!!.visibility = View.GONE
                activity!!.toast("Error al enviar el codigo.")
            }
        })

    }
    fun accountActivate(e1: String) {
        progressBar_correo!!.visibility = View.VISIBLE
        val sendCorreo =
            RetrofitClient.getInstance().api.sendEmail(e1);
        sendCorreo.enqueue(object: Callback<Driver>{
            override fun onFailure(call: Call<Driver>, t: Throwable) {
                progressBar_correo!!.visibility = View.GONE
                Log.e("accountActivate",t.message)
                activity!!.toast("Error con la red.")
            }

            override fun onResponse(call: Call<Driver>, response: Response<Driver>) {
                Log.e("accountActivate",response.code().toString())
                when {
                    response.code()==200 -> {
                        // setIDrecuperar(navFragment, response.body()!!.Driver)
                        setNavFragment(navFragment, correo().toString())
                        setCorreoNavFragment(navFragment, e1)
                        val bundle= Bundle()
                        bundle.putBoolean( "validar_account",true)
                        bundle.putBoolean("activity" ,true)
                        findNavController().navigate(R.id.action_correo_to_set_codigo,bundle)
                    }
                    response.code()==202 -> {
                        progressBar_correo!!.visibility = View.GONE
                        activity!!.toast("Correo ya esta siendo usado")
                    }
                    else -> {
                        progressBar_correo!!.visibility = View.GONE
                        activity!!.toast("Intentelo nuevamente")
                    }
                }
            }

        })

    }
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {

        return inflater.inflate(R.layout.fragment_correo, container, false)
    }

    private fun comprobarcampos(e1: String, e2: String): Boolean {
        if (e1!!.isEmpty() || e2!!.isEmpty()) {
            c_edtxt_email!!.error = "Campos vacios"
            c_edtxt_email_confirm!!.error = "Campos vacios"
            return false
        }
        if (!Patterns.EMAIL_ADDRESS.matcher(e1).matches()) {
            c_edtxt_email!!.error = "No es un correo valido"
            return false
        }
        if (!Patterns.EMAIL_ADDRESS.matcher(e2).matches()) {
            c_edtxt_email_confirm!!.error = "No es un correo valido"
            return false
        }
        if (e1 != e2) {
            c_edtxt_email!!.error = "Los correos no coinciden"
            c_edtxt_email_confirm!!.error = "Los correos no coinciden"
            return false
        }
        return true
    }
}